velox/functions/prestosql/Map.cpp

```diff
@@ -213,7 +213,8 @@ class MapFunction : public exec::VectorFunction {
     VELOX_CHECK_GE(keys->size(), rows.end());
     VELOX_CHECK_GE(values->size(), rows.end());
     return rows.testSelected([&](vector_size_t row) {
-      return keys->offsetAt(row) == values->offsetAt(row);
+      return keys->offsetAt(row) == values->offsetAt(row) &&
```

Comment:
The comment needs updating. Currently, it suggests that it is not checking for sizes intentionally.

```
  // (if element sizes are different keys and values the Map function will
  // throw).
```

Also, if canTakeFastPath checks for sizes, then this branch should not check sizes again. That makes me thing that the problem is somewhere else. I wonder if the issue manifests itself only when running under TRY in which case some rows may report an error, but we continue processing these. I fixed a similar bug in https://github.com/facebookincubator/velox/pull/3514 . Perhaps, the following assumption is wrong:

```
      // In the fast path, we do not need to exclude 'failed' rows from
      // processing. It is same to include these in the MapVector and rely on
      // the expression evaluation framework to handle failures.
```